executor/infoschema_reader.go

```diff
@@ -49,6 +49,20 @@ import (
 	"github.com/pingcap/tidb/util/stmtsummary"
 )
 
+func newMemTableRetriever(table *model.TableInfo, columns []*model.ColumnInfo) memTableRetriever {
+	switch table.Name.O {
+	case infoschema.TableColumns:
+		return &hugeMemTableRetriever{
+			table:   table,
+			columns: columns,
+		}
+	}
```

Comment:
I think it's duplicated. Why not move this `switch-case` into `executor/builder.go` where you moved it out?